sdk/identity/Azure.Identity/src/DefaultAzureCredentialFactory.cs

```diff
@@ -151,6 +141,116 @@ public TokenCredential[] CreateCredentialChain()
             return chain.ToArray();
         }
 
+        /// <summary>
+        /// Validates and retrieves the value from a custom environment variable.
+        /// Throws InvalidOperationException if the variable is not set or is empty.
+        /// </summary>
+        /// <param name="environmentVariableName">The name of the environment variable to validate and retrieve.</param>
+        /// <returns>The trimmed and lowercased environment variable value.</returns>
+        private static string ValidateAndGetCustomEnvironmentVariable(string environmentVariableName)
+        {
+            // Validate environment variable name contains only letters, digits, or underscores
+            for (int i = 0; i < environmentVariableName.Length; i++)
+            {
+                char c = environmentVariableName[i];
+                if (!char.IsLetterOrDigit(c) && c != '_')
+                {
+                    throw new ArgumentException($"Invalid environment variable name: '{environmentVariableName}'. Only letters, digits, and underscores are allowed.", nameof(environmentVariableName));
+                }
+            }
+
+            string credentialSelection = Environment.GetEnvironmentVariable(environmentVariableName);
+            if (string.IsNullOrEmpty(credentialSelection))
+            {
+                throw new InvalidOperationException($"Environment variable '{environmentVariableName}' is not set or is empty.");
+            }
+            return credentialSelection.Trim().ToLower();
+        }
+
+        /// <summary>
+        /// Processes a credential selection value and returns the appropriate credential chain.
+        /// </summary>
+        /// <param name="credentialSelection">The credential selection value (already trimmed and lowercased).</param>
+        /// <param name="environmentVariableName">The environment variable name for error messages.</param>
+        /// <returns>Array of TokenCredential instances based on the selection.</returns>
+        private TokenCredential[] ProcessCredentialSelection(string credentialSelection, string environmentVariableName)
+        {
+            bool useDevCredentials = Constants.DevCredentials.Equals(credentialSelection, StringComparison.OrdinalIgnoreCase);
+            bool useProdCredentials = Constants.ProdCredentials.Equals(credentialSelection, StringComparison.OrdinalIgnoreCase);
+
+            if (useDevCredentials)
+            {
+                return CreateDevelopmentCredentialChain();
+            }
+            else if (useProdCredentials)
+            {
+                return CreateProductionCredentialChain();
+            }
+            else
+            {
+                return CreateSpecificCredentialChain(credentialSelection, environmentVariableName);
+            }
```

Comment:
Nit: A switch expression modernizes this code block and cuts the number of lines in half.


```suggestion
            return (useDevCredentials, useProdCredentials) switch
            {
                (true, _) => CreateDevelopmentCredentialChain(),
                (_, true) => CreateProductionCredentialChain(),
                _ => CreateSpecificCredentialChain(credentialSelection, environmentVariableName)
            };
```